### PR TITLE
[move-prover] Adds cvc4 to tests

### PR DESCRIPTION
**Pull request by barrettcw**
This adds a set of unit tests to be run with cvc4 when running "cargo test"

### CODE REVIEW

File: language/move-prover/tests/testsuite.rs
Comment (wrwg) [inline language/move-prover/tests/testsuite.rs:103]: This could be done nicer if you like so. See the use of extract_test_directives in the `let baseline_valid = ...`. You could have `!extract_test_directive("no-cvc4")?` and then just have a comment `// no-cvc4` + explanations why in the source of the test. But I leave it to you, not sure how much time pressure there is for getting this done.

**Review by wrwg — APPROVED** [review]

File: language/move-prover/tests/testsuite.rs
Comment (DavidLDill) [reply language/move-prover/tests/testsuite.rs:103]: Clark can't land this, I think, because his affiliation with Facebook ended.  So, I'm going to try to land it and we can fix this up in a future PR.

Comment (wrwg) [reply language/move-prover/tests/testsuite.rs:103]: The PR workflow did not pass. I tried to rerun it yesterday because the failure looked like a flake, but it again failed. Otherwise I would have already landed.

**Pull request closed without merging**
